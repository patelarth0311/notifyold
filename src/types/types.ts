export type Note = {
    name: string,
    content: Entry[],
    noteId: string,
}



export type FilterOption = {
    img: string,
    action: () => void
}


export type Model = {
    open: boolean
    setOpen: (e: boolean) => void
}

export type Entry = {
    entryId?: string,
    content: string,
    flag: boolean,
    priority: number,
    time: string
}

export type  AppStatus = {
    status: string
    response: string
    userId: string | null
  
}

export type UserFormFields = {
    email: string,
    password: string,
    hasConfirmPassword: boolean
    confirmPassword?: string
    buttonText: string
    messageText: string
    optionText: string
    setShow: () => void
    buttonAction: () => void
    setForm: (e: any) => void
}

export type User = {
    userId: string
    username: string
    status: boolean
}