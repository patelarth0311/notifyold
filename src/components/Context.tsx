import { Dispatch, SetStateAction, createContext } from "react";
import { AppStatus } from "../types/types";
export const MyContext = createContext<{ appStatus: AppStatus, setAppStatus: (e: AppStatus) => void }>({
    appStatus: { response: "", status: "", userId: null },
    setAppStatus: (e: AppStatus) => { }
});