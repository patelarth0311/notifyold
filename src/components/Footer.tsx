import github from "../svgs/github.svg";
import linkedin from "../svgs/linkedin.svg";

export function Footer() {
  return (
    <div className="flex flex-col  bg-indigo-600 justify-center gap-y-3 items-center h-[140px]">
      <h1 className="text-xl font-ver text-center text-white">Notify</h1>
      <div className="flex  gap-x-4 ">
        <a
          target="_blank"
          rel="noopener noreferrer"
          href="https://github.com/patelarth0311"
        >
          <img width={25} src={github}></img>
        </a>
        <a
          target="_blank"
          rel="noopener noreferrer"
          href="https://www.linkedin.com/in/patelrarth/"
        >
          <img width={25} src={linkedin}></img>
        </a>
      </div>
    </div>
  );
}
