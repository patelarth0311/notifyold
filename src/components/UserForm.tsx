import { User, UserFormFields } from "../types/types";
import { PinInput, PinInputField } from "@chakra-ui/react";
import { HStack } from "@chakra-ui/react";
import { useState } from "react";
import { confirmUser } from "../API/Requests";

export function UserForm(
  props: UserFormFields & { showErrorMessage: boolean }
) {
  const [interacted, setInteracted] = useState(false);

  return (
    <div>
      <div className="flex font-ver justify-center items-center flex-col gap-y-20 w-full ">
        <div className="flex justify-center items-center flex-col gap-y-16 ">
          <div className="relative">
            <p className="absolute top-[-30px] text-[14px] ">Email</p>

            <input
              name="email"
              value={props.email}
              onChange={props.setForm}
              title="Email"
              type="email"
              className="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
            ></input>
          </div>

          <div className="relative">
            <p
              className={
                "absolute top-[-30px] text-[14px] " +
                (props.showErrorMessage ? "text-red-600" : "")
              }
            >
              Password
            </p>
            <input
              name="password"
              value={props.password}
              onFocus={(e) => {
                setInteracted(true);
              }}
              onChange={props.setForm}
              title="Password"
              type="password"
              className="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
            ></input>
          </div>
          {props.hasConfirmPassword && (
            <div className="relative flex flex-col gap-y-3">
              <p
                className={
                  "absolute top-[-30px] text-[14px] " +
                  (props.showErrorMessage ? "text-red-600" : "")
                }
              >
                Confirm Password
              </p>
              <input
                disabled={props.password.length === 0}
                name="confirmPassword"
                value={props.confirmPassword}
                onChange={props.setForm}
                title="Password"
                type="password"
                className={
                  "w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border " +
                  (props.password.length === 0 ? "bg-slate-100" : "")
                }
              ></input>
              {(props.showErrorMessage || interacted) && (
                <div className="text-[12px]  w-full justify-start  flex flex-col items-start h-full">
                  <p className="text-[13px]">Password must contain:</p>

                  <p>At least one lowercase and uppercase letter</p>
                  <p>At least one special character</p>
                  <p>At least one number</p>
                  <p>Be at least 8 characters</p>
                </div>
              )}
            </div>
          )}
        </div>

        <div className="flex flex-col gap-y-5">
          <button
            onClick={props.buttonAction}
            className="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
          >
            {props.buttonText}
          </button>

          <div className="flex w-[353px]  items-center justify-end gap-x-3 ">
            <p>{props.messageText}</p>
            <button
              onClick={props.setShow}
              className="text-base text-indigo-500 font-ver"
            >
              {props.optionText}
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

export function ConfirmationPin(props: { user: User; register: () => void }) {
  return (
    <div className="font-ver relative">
      <p className="absolute top-[-30px] text-[14px]">Enter code</p>
      <HStack>
        <PinInput
          onComplete={(e) => {
            confirmUser(props.user.username, e).then((res) => {
              console.log(res);
              if (res && res.status === 200) {
                props.register();
              }
            });
          }}
          type="number"
          placeholder=""
        >
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px] " />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
          <PinInputField className="w-[50px] border h-[50px] rounded-[10px] pl-[19px]" />
        </PinInput>
      </HStack>
    </div>
  );
}
