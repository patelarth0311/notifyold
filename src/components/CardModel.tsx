import { Entry, Model, Note, AppStatus } from "../types/types";
import { useEffect, useState } from "react";
import { useRef } from "react";
import close from "../svgs/close.svg";
import edit from "../svgs/edit.svg";
import plus from "../svgs/plus.svg";
import {
  deleteNote,
  postEntry,
  deleteEntry,
  changeEntry,
} from "../API/Requests";
import cal from "../svgs/calendar.svg";
import priority from "../svgs/priority.svg";
import { useContext } from "react";
import { MyContext } from "./Context";
import "../App.css";
import trash from "../svgs/trash.svg";
import { useToast } from "@chakra-ui/react";
import check from "../svgs/check.svg";
import minus from "../svgs/minus.svg";
import { makeToast } from "./Toast";
import flag from "../svgs/flag.svg"

export function CardModel(props: Model & { note: Note }) {
  const [editMode, setEditMode] = useState(false);
  const toast = useToast();
  const { appStatus, setAppStatus } = useContext(MyContext);
  const [createEntry, setCreateEntry] = useState(false);

  const [editEntry, setEditEntry] = useState({
    content: "",
    flag: false,
    time: "",
    priority: 0,
  });

  const [entry, setEntry] = useState({
    content: "",
    flag: false,
    time: "",
    priority: 0,
  });

  const handleChange = (e: any) => {
    if (e.target) {
      const { name, value, type, checked } = e.target;
    
      setEntry((prev) => {
        return {
          ...prev,
          [name]: type === "checkbox" ? checked : value,
        };
      });
    }
  };

  return (
    <div className="flex gap-y-4 flex-col p-10 ">
      <div className="flex gap-x-3 justify-end ">
        <button
          onClick={() => {
            deleteNote(
              props.note.noteId,
              appStatus.userId ? appStatus.userId : ""
            ).then((res) => {
              if (res?.status === 200) {
                setAppStatus({
                  ...appStatus,
                  status: res.data.status,
                  response: res.data,
                });
                props.setOpen(false);
                makeToast(  <img className="absolute left-10" src={check}></img>,props.note.name,"border-[#6FCF97]",toast, "deleted!")
              }
            });
          }}
          className="shadow-2xl rounded-full bg-indigo-500 w-8 h-8  flex justify-center items-center "
        >
          <img width={15} src={trash}></img>
        </button>

        {props.note.content.length > 0 && (
          <button
            onClick={() => {
              setEditMode((prev) => !prev);
              setCreateEntry(false);
            }}
            className={
              "shadow-2xl rounded-full bg-indigo-500 w-8 h-8 flex justify-center items-center " +
              (editMode ? " animate-bounce " : "")
            }
          >
            <img width={15} src={edit}></img>
          </button>
        )}

        <button
          onClick={() => props.setOpen(false)}
          className="shadow-2xl rounded-full bg-indigo-500 w-8 h-8  flex justify-center items-center "
        >
          <img width={15} src={close}></img>
        </button>
      </div>

      <div
        className={
          "mobile:w-[355px]  relative shadow-custom mobile:h-[410px]   w-[493.5px] h-[517.5px]   aspect-square rounded-[21.67px] bg-white "
        }
      >
        <div className="flex justify-start p-[15px] flex-col h-full b">
          <div className="justify-start flex  h-full gap-y-2 flex-col ">
            <p className="text-[17px] text-left font-bold text-black sticky">
              {props.note.name}
            </p>
            <div className="overflow-y-auto h-full relative ">
              <div className="flex justify-start items-start flex-col overflow-hidden w-full gap-y-3 overflow-y-auto p-1">
                {props.note.content.map((item, index) => (
                  <EditEntryField
                    appStatus={appStatus}
                    setAppStatus={setAppStatus}
                    noteId={props.note.noteId}
                    key={item.entryId}
                    item={item}
                    editMode={editMode}
                    editEntry={editEntry}
                    setEditEntry={(e: Entry) => {
                      setEditEntry(e);
                    }}
                  ></EditEntryField>
                ))}
                {createEntry && (
                  <CreateEntryField
                    entry={entry}
                    handleChange={handleChange}
                  ></CreateEntryField>
                )}
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className=" flex justify-between items-center w-full">
        <button
          onClick={() => {
            setEditMode(false);
            setCreateEntry((prev) => !prev);
          }}
          className={
            "  shadow-2xl rounded-full bg-indigo-500 w-8 h-8 flex justify-center items-center "
          }
        >
          <img width={15} src={plus}></img>
        </button>

        {(createEntry || editMode) && (
          <div className=" flex gap-x-1">
            {(props.note.content.length > 0 || createEntry) && (
              <button
                onClick={() => {
                  setEditMode(false);
                  setCreateEntry(false);
                  setEntry({ content: "", time: "", priority: 0, flag: false });
                }}
                className={
                  "bg-indigo-500 w-[90px] h-[30px] rounded-md text-white "
                }
              >
                Cancel
              </button>
            )}

            {createEntry && (
              <button
                onClick={() => {
                  try {
                    postEntry(
                      entry,
                      props.note.noteId,
                      appStatus.userId ? appStatus.userId : ""
                    ).then((res) => {
                      if (res?.status === 200) {
                        setAppStatus({
                          ...appStatus,
                          status: res.data.status,
                          response: res.data,
                        });
                        setEditMode(false);
                        setCreateEntry(false);
                        setEntry({
                          content: "",
                          time: "",
                          priority: 0,
                          flag: false,
                        });
                      }
                    });
                  } catch (err) {}
                }}
                className={
                  "bg-indigo-500 w-[90px] h-[30px]  rounded-md text-white "
                }
              >
                Submit
              </button>
            )}
          </div>
        )}
      </div>
    </div>
  );
}
function CreateEntryField(props: {
  entry: Entry;
  handleChange: (e: any) => void;
}) {
  return (
    <div className="flex flex-col gap-y-2 w-full">
      <input
        type="text"
        name="content"
        value={props.entry.content}
        onChange={props.handleChange}
        className="text-[15px] border p-1   relative rounded-md  "
        placeholder="Add info"
      ></input>
      <div className="relative flex w-full gap-x-10 ">
        <div className="flex gap-x-2  w-full">
          <img src={cal} width={20}></img>
          <input
            name="time"
            value={props.entry.time}
            onChange={props.handleChange}
            type="date"
            className=" border  w-full rounded-lg focus:outline-indigo-500 pl-1 p-1"
            placeholder="Select date"
          />
        </div>
        <div className="flex gap-x-2">
          <img width={20} src={priority}></img>

          <select
          defaultValue={props.entry.priority}
          
            onChange={props.handleChange}

            name="priority"
            placeholder="Select Priority"
          >
            <option value="0">None</option>
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
          </select>
        </div>

        <label className="flex gap-x-2">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            fill={props.entry.flag ? "#84cc16" : "#d1d5db"}
            className="w-14"
          >
            <path
              fillRule="evenodd"
              d="M3 2.25a.75.75 0 01.75.75v.54l1.838-.46a9.75 9.75 0 016.725.738l.108.054a8.25 8.25 0 005.58.652l3.109-.732a.75.75 0 01.917.81 47.784 47.784 0 00.005 10.337.75.75 0 01-.574.812l-3.114.733a9.75 9.75 0 01-6.594-.77l-.108-.054a8.25 8.25 0 00-5.69-.625l-2.202.55V21a.75.75 0 01-1.5 0V3A.75.75 0 013 2.25z"
              clipRule="evenodd"
            />
          </svg>

          <input
            value={props.entry.flag.toString()}
            type="checkbox"
            name="flag"
            onClick={props.handleChange}
            className="w-full h-full peer appearance-none rounded-md"
          />
        </label>
      </div>
    </div>
  );
}

function EditEntryField(props: {
  item: Entry;
  editMode: boolean;
  editEntry: Entry;
  setEditEntry: (e: Entry) => void;
  noteId: string;
  setAppStatus: (e: AppStatus) => void;
  appStatus: AppStatus;
}) {
  const [selectedField, setSelectedField] = useState(false);
  const inputRef = useRef<HTMLInputElement>(null);
  const [editEntry, setEditEntry] = useState({
    content: props.item.content,
    flag: props.item.flag,
    time: props.item.time,
    priority: props.item.priority,
  });

  useEffect(() => {

    const handleInput  = (e: Event) => {
      if (inputRef.current) {
        if (!inputRef.current.contains(e.target as Node)) {
          setSelectedField(false);
          if (!inputRef.current.contains(e.target as Node)) {
            setSelectedField(false);
          }
        }
      }
    }

    document.addEventListener("click", (e) => {
        handleInput(e)
    });

    return () => document.removeEventListener("click", (e) => {
        handleInput(e)
    });
  }, []);

  useEffect(() => {
    if (selectedField === false) {
      setEditEntry(props.item);
    }
  }, [selectedField]);

  return (
    <div className="flex flex-row gap-x-2 justify-between w-full ">
      <div className="w-full">
      <input
        ref={inputRef}
        value={editEntry.content}
        onClick={() => {
          setSelectedField((prev) => !prev);
        }}
        onKeyDown={(e) => {
          if (props.item.entryId && e.key === "Enter") {
            changeEntry(
              editEntry,
              props.noteId,
              props.appStatus.userId ? props.appStatus.userId : ""
            ).then((res) => {
              if (res?.status === 200) {
                props.setAppStatus({
                  ...props.appStatus,
                  status: res.data.status,
                  response: res.data,
                });
              }
            });
          }
        }}
        onChange={(e) => {
          setEditEntry((prev) => {
            return { ...prev, content: e.target.value };
          });
        }}
        className={
          "text-[15px] h-auto bg-white focus:outline-none relative rounded-md  w-full p-1 " +
          (props.editMode ? "  border" : "")
        }
        disabled={!props.editMode}
      ></input>
         <div className="flex  gap-x-1 p-1 items-center justify-start ">
            <p className="text-[11px] font-medium text-black text-left">{props.item.time}</p>
            <p className="text-[11px] font-medium  text-left text-[#f59e0b]">{"!".repeat(props.item.priority)}</p>
            {props.item.flag && (
              <img width={12} src={flag}></img>
            )
            }
              </div>
      </div>


      {props.editEntry && selectedField && (
        <button
          onClick={() => {
            if (props.item.entryId) {
              deleteEntry(
                props.noteId,
                props.item.entryId,
                props.appStatus.userId ? props.appStatus.userId : ""
              ).then((res) => {
                if (res?.status === 200) {
                  props.setAppStatus({
                    ...props.appStatus,
                    status: res.data.status,
                    response: res.data,
                  });
                }
              });
            }
          }}
          className="w-7 h-7 justify-center items-center"
        >
          <img height={2} src={minus}></img>
        </button>
      )}
    </div>
  );
}
