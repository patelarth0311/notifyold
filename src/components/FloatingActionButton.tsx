import menu from "../svgs/menu.svg";
import pencil from "../svgs/pencil.svg";
import logout from "../svgs/signout.svg";
import { useState } from "react";
import { ListCreation } from "./ListCreation";
import close from "../svgs/close.svg";
import information from "../svgs/information.svg"
import { MyContext } from "./Context";
import { useContext } from "react";

export function FloatingActionButton() {
  const [toggle, setToggle] = useState(false);

  return (
    <div className="fixed font-ver z-50">
      <button
        onClick={() => {
          setToggle((prev) => !prev);
        }}
        className="shadow-2xl rounded-full bg-indigo-500 w-12 h-12 flex justify-center items-center fixed top-6 right-6"
      >
        <img width={24} src={toggle ? close : menu}></img>
      </button>
      {toggle && (
        <>
          <Create fixed={true}></Create>
          <Info></Info>
          <Logout></Logout>
        </>
      )}
    </div>
  );
}
export function Create(props: {fixed: boolean}) {
  const [toggle, setToggle] = useState(false);

  return (
    <>
      <button
        onClick={() => {
          setToggle((prev) => !prev);
          document.body.style.overflow = "hidden";
        }}
        className={"shadow-2xl  rounded-full bg-indigo-500 w-10 h-10 flex justify-center items-center " + (props.fixed ? " fixed top-24 right-7 " : "relative")}
      >
        <img width={20} src={pencil}></img>
      </button>
      {toggle && (
        <ListCreation
          setToggle={() => {
            document.body.style.overflowY = "scroll";
            setToggle(false);
          }}
        ></ListCreation>
      )}
    </>
  );
}

function Logout() {

    const {appStatus, setAppStatus} = useContext(MyContext)

  return (
    <button
      onClick={() => {
        localStorage.clear();
        setAppStatus({userId: "", response: "", status: ""})

      }}
      className="shadow-2xl rounded-full bg-indigo-500 w-10 h-10 flex justify-center items-center fixed top-7 right-24"
    >
      <img width={20} src={logout}></img>
    </button>
  );
}

function Info() {
  return (
    <button className="shadow-2xl rounded-full bg-indigo-500 w-10 h-10 flex justify-center items-center fixed top-20 right-20">
      <img width={20} src={information}></img>
    </button>
  );
}
