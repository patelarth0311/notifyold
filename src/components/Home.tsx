import { ToolBar } from "./ToolBar";
import { Create, FloatingActionButton } from "./FloatingActionButton";
import search from "../svgs/search.svg";
import { Entry, FilterOption, Note } from "../types/types";
import { useEffect, useState } from "react";
import { CardModel } from "./CardModel";
import { OptionsBar } from "./OptionsBar";
import { Lists } from "./Lists";
import "../App.css";
import { readNote } from "../API/Requests";
import { useContext } from "react";
import { MyContext } from "./Context";
import error from "../svgs/error.svg";
import {
  CognitoUser,
  CognitoAccessToken,
  CognitoIdToken,
  CognitoRefreshToken,
  CognitoUserSession,
} from "amazon-cognito-identity-js";
import { makeToast } from "./Toast";

import { useToast } from "@chakra-ui/react";

var checkSession = () => {
  const idToken = localStorage.getItem("IdToken");
  const accessToken = localStorage.getItem("AccessToken");
  const refreshToken = localStorage.getItem("RefreshToken");
  if (idToken && accessToken && refreshToken) {
    const AccessToken = new CognitoAccessToken({
      AccessToken: accessToken,
    });
    const IdToken = new CognitoIdToken({
      IdToken: idToken,
    });

    const RefreshToken = new CognitoRefreshToken({
      RefreshToken: refreshToken,
    });

    const sessionData = {
      IdToken: IdToken,
      AccessToken: AccessToken,
      RefreshToken: RefreshToken,
    };

    const cachedSession = new CognitoUserSession(sessionData);
    console.log(cachedSession)
    if (cachedSession.isValid()) {
      return true;
    }

    if (!RefreshToken.getToken()) {
      return false;
    }
  }
  return false;
};

export function Home() {
  const toast = useToast();
  const { appStatus, setAppStatus } = useContext(MyContext);

  return (
    <div className="flex min-h-screen width-full justify-end text-center 0  relative flex-col  bg-neutral-50">
      <div className="min-h-[180px] flex flex-col justify-center">
        <p className="font-ver text-4xl  text-center ">Notify</p>
      </div>

      <NoteLibrary></NoteLibrary>
      <FloatingActionButton></FloatingActionButton>
    </div>
  );
}

export function NoteLibrary() {
  const [notes, setNote] = useState<Note[]>([]);
  const toast = useToast()
  const { appStatus, setAppStatus } = useContext(MyContext);
  const [open, setOpen] = useState(false);
  const [optionFilter, setOptionFilter] = useState("");
  const [searchText, setSearchText] = useState("");

  const [noteSelect, setNoteSelected] = useState({
    name: "",
    content: [] as Entry[],
    noteId: "",
  });

  useEffect(() => {
    if (!checkSession()) {
      {makeToast(   <img className="absolute left-10" width={20} src={error}></img>,"You have been signed out","border-[#dc2626]",toast)}
      localStorage.clear()
      setAppStatus({userId: "", response: "", status: ""})
    }
  },[])


  useEffect(() => {

    if (appStatus.userId) {
      readNote(appStatus.userId).then((res) => {
     
        if (res) {
          if (res.data.Items) {
            setNote(
              res.data.Items.map((item: Note, index: number) => {
                if (noteSelect.noteId === item.noteId) {
                  setNoteSelected({
                    name: item.name,
                    content: Object.values(item.content),
                    noteId: item.noteId,
                  });
                }
               
                return {
                  name: item.name,
                  content: Object.values(item.content),
                  noteId: item.noteId,
                };
              })
            );
          }
        }
      });
    }




  }, [appStatus]);





  return (
    <div className="flex  grow   font-ver   flex-col items-center gap-y-20">
      <SearchBar
        searchText={searchText}
        setSearchText={(e: any) => {
          if (e.currentTarget) {
            setSearchText(e.currentTarget.value);
          }
        }}
      ></SearchBar>
      <OptionsBar
        optionFilter={optionFilter}
        setOptionFilter={(e: string) => {
          setOptionFilter(e);
        }}
      ></OptionsBar>
      <div className="flex   justify-end  grow flex-col items-center   gap-y-20  ">
        {notes.length > 0 ? (
          <>
            {open ? (
              <div className="flex flex-col justify-center  items-center h-full w-full   ">
                <CardModel
                  note={noteSelect}
                  open={open}
                  setOpen={(e: boolean) => setOpen(e)}
                ></CardModel>

                <div id="duo-container" className=" relative">
                  <div className="   flex flex-row  pl-10  gap-5 pr-10 pb-10 justify-start  overflow-y-auto   w-screen    ">
                    {notes
                      .filter((item) => {
                        const metConditionList = item.content.map((item) => {
                          switch (optionFilter) {
                            case "priority":
                                return item.priority !== 0;
                              case "flagged":
                                return item.flag === true;
                              case "date":
                                return item.time !== undefined;
                            default:
                              return true;
                          }
                        });
                        return metConditionList.includes(true);
                      })
                      .filter((item) => {
                        return item.name
                          .toLowerCase()
                          .includes(searchText.toLowerCase());
                      })
                      .map((item, index) => (
                        <Card
                          note={item}
                          action={() => {
                            setNoteSelected(item);
                          }}
                          key={index}
                        ></Card>
                      ))}
                  </div>
                </div>
              </div>
            ) : (
              <div className="grid grow grid-cols-2 mobile2:grid-cols-1 gap-5 pb-10   gap-y-12 place-content-start">
                {notes
                  .filter((item) => {
                    const metConditionList = item.content.map((item) => {
                     
                      switch (optionFilter) {
                        case "priority":
                          return item.priority !== 0;
                        case "flagged":
                          return item.flag === true;
                        case "date":
                          
                          return item.time !== undefined;
                        default:
                          return true;
                      }
                    });
         
                    return (item.content.length === 0 && optionFilter === "") || metConditionList.includes(true);
                  })
                  .filter((item) => {
                    return item.name
                      .toLowerCase()
                      .includes(searchText.toLowerCase());
                  })
                  .map((item, index) => (
                    <Card
                      note={item}
                      action={() => {
                        setOpen(true);
                        setNoteSelected(item);
                      }}
                      key={index}
                    ></Card>
                  ))}
              </div>
            )}
          </>
        ) : (
        
          <div className=" grow flex flex-col items-center gap-y-10 ">
          <p className="text-3xl font-ver ">No notes ðŸ«¨</p>
          <Create fixed={false}></Create>
          </div>

        )}
      </div>
    </div>
  );
}

export function Card(props: { action: () => void; note: Note }) {
  return (
    <div
      onClick={() => {
        props.action();
      }}
      className={
        "min-w-[338px] w-[338px] min-h-[155px] max-h-[157px] h-[157px] shadow-custom rounded-[21.67px] bg-white "
      }
    >
      <div className="flex justify-start p-[16px] gap-x-5 h-full">
        <div className=" justify-start items-start flex flex-col">
          <p className="text-[17px] font-bold ">{props.note.name}</p>
          <p className="text-[24px] font-bold ">{props.note.content.length}</p>
        </div>
        <div className="flex overflow-y-auto relative h-full flex-col w-full gap-y-1 text-gray-500">
          {props.note.content.map((item, index) => (
            <p className="text-[15px] text-left" key={index}>
              {item.content}
            </p>
          ))}
        </div>
      </div>
    </div>
  );
}

export function SearchBar(props: {
  searchText: string;
  setSearchText: (e: any) => void;
}) {
  return (
    <div className="flex relative justify-start gap-x-3  h-[44px] w-[350px] shadow-xl bg-white rounded-lg p-2">
      <img width={15} src={search}></img>

      <input
        value={props.searchText}
        onChange={(e) => {
          props.setSearchText(e);
        }}
        className="bg-white focus:outline-none  rounded-md w-full"
        placeholder="Search"
        type="text"
      ></input>
    </div>
  );
}
