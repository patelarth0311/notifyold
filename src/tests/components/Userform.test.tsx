import { cleanup, getByText, render, fireEvent, act, waitFor} from '@testing-library/react';
import { UserForm } from '../../components/UserForm';
import { UserFormFields } from '../../types/types';


describe('forms',  () => { 

    const data  = {
        email: 'random03@gmail.com',
        password: 'abc123',
        hasConfirmPassword: false,
        buttonText: '',
        messageText: '',
        optionText: '',
        setShow: function (): void {
            throw new Error('Function not implemented.');
        },
        buttonAction: function (): void {
            throw new Error('Function not implemented.');
        },
        setForm: function (e: any): void {
            throw new Error('Function not implemented.');
        },
        showErrorMessage: false
    }
    const form =  render(<UserForm {...data}></UserForm>)

    test('Check if email passed', async () => {
        expect(await form.findByRole('textbox', {name: 'email'})).toHaveValue("random03@gmail.com")

    })


    expect(form).toMatchSnapshot()

    

 

 })