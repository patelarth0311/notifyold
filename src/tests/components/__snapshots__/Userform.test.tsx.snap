// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex font-ver justify-center items-center flex-col gap-y-20 w-full "
        >
          <div
            class="flex justify-center items-center flex-col gap-y-16 "
          >
            <div
              class="relative"
            >
              <p
                class="absolute top-[-30px] text-[14px] "
              >
                Email
              </p>
              <input
                aria-label="email"
                class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                name="email"
                role="textbox"
                title="Email"
                type="text"
                value="random03@gmail.com"
              />
            </div>
            <div
              class="relative"
            >
              <p
                class="absolute top-[-30px] text-[14px] "
              >
                Password
              </p>
              <input
                aria-label="password"
                class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
                name="password"
                title="Password"
                type="password"
                value="abc123"
              />
            </div>
          </div>
          <div
            class="flex flex-col gap-y-5"
          >
            <button
              aria-label="process-user-button"
              class="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
            />
            <div
              class="flex w-[353px]  items-center justify-end gap-x-3 "
            >
              <p />
              <button
                aria-label="registrationoption"
                class="text-base text-indigo-500 font-ver"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex font-ver justify-center items-center flex-col gap-y-20 w-full "
      >
        <div
          class="flex justify-center items-center flex-col gap-y-16 "
        >
          <div
            class="relative"
          >
            <p
              class="absolute top-[-30px] text-[14px] "
            >
              Email
            </p>
            <input
              aria-label="email"
              class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
              name="email"
              role="textbox"
              title="Email"
              type="text"
              value="random03@gmail.com"
            />
          </div>
          <div
            class="relative"
          >
            <p
              class="absolute top-[-30px] text-[14px] "
            >
              Password
            </p>
            <input
              aria-label="password"
              class="w-[353px] pl-[10px] pr-[10px] rounded-[10px] h-[56px] border"
              name="password"
              title="Password"
              type="password"
              value="abc123"
            />
          </div>
        </div>
        <div
          class="flex flex-col gap-y-5"
        >
          <button
            aria-label="process-user-button"
            class="w-[353px] h-[56px] pl-[10px] pr-[10px] rounded-[10px] font-ver bg-indigo-500 text-white text-base"
          />
          <div
            class="flex w-[353px]  items-center justify-end gap-x-3 "
          >
            <p />
            <button
              aria-label="registrationoption"
              class="text-base text-indigo-500 font-ver"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
